cse-runner-app/src/main/java/com/farao_community/farao/cse/runner/app/util/MerchantLine.java

```diff
@@ -59,7 +76,52 @@ private static void activateMerchantLineForIdcc(Network network) {
             MENDRISIO_ID, phaseTapChanger.getRegulationValue()));
     }
 
-    private static void activateMerchantLineForD2cc(Network network) {
-        // TODO: Implement merchant line handling for D2CC
+    private static void activateMerchantLineForD2cc(Network network, CseData cseData) {
+        runLoadFlow(network);
+        double defaultFlow = cseData.getNtc().getFlowOnFixedFlowLines().get(MENDRISIO_CAGNO_CODE_IN_NTC_FILE);
+        UcteNetworkAnalyzer ucteNetworkHelper = new UcteNetworkAnalyzer(network, new UcteNetworkAnalyzerProperties(UcteNetworkAnalyzerProperties.BusIdMatchPolicy.COMPLETE_WITH_WILDCARDS));
+        Optional<Double> reducedFlow = cseData.getLineFixedFlows().getFixedFlow(MENDRISIO_CAGNO_CODE_IN_TARGET_CH_FILE, network, ucteNetworkHelper);
+        double mendrisioCagnoTargetFlow = reducedFlow.isEmpty() ? defaultFlow : Math.min(defaultFlow, reducedFlow.get());
+
+        LOGGER.info(String.format("Target flow for Mendrisio-Cagno is %.0f MW", mendrisioCagnoTargetFlow));
+
+        // Offset Calculation
+        TwoWindingsTransformer mendrisioTransformer = network.getTwoWindingsTransformer(MENDRISIO_ID);
+        if (mendrisioTransformer == null) {
+            throw new CseInvalidDataException(String.format(
+                    "Mendrisio transformer is not present in the network with the following ID : %s", MENDRISIO_ID));
+        }
+        PhaseTapChanger phaseTapChanger = mendrisioTransformer.getPhaseTapChanger();
+        if (phaseTapChanger == null) {
+            throw new CseInvalidDataException(String.format(
+                    "Mendrisio transformer (%s) has no phase tap changer", MENDRISIO_ID));
+        }
+
+        double mendrisioPstFlow = phaseTapChanger.getRegulationValue();
```

Comment:
You should not compare with the regulation value but the actual flow on the PST with getTermial1().getP().